File: 프로그래머스/0/120894. 영어가 싫어요/영어가 싫어요.java
class Solution {
    public long solution(String numbers) {
        long ret = 0;
        String[] num = { "zero", "one", "two", "three", "four", "five", "six", "seven", "eight", "nine"};
        String temp = "";
        String temp1 = "";
        for(char c : numbers.toCharArray()){
            temp += c;
            if(temp.length() >= 3){
                for(int i = 0; i <= 9; i++){
                    if(temp.equals(num[i])){
                        temp1 += i;
                        temp = "";
                    }
                }
            }
        }
        ret = Long.parseLong(temp1);
        
        
        return ret;
    }
}

File: 프로그래머스/0/120894. 영어가 싫어요/영어가 싫어요.py
def solution(numbers):
    dic ={"zero":'0', "one":'1', "two":'2', "three":'3', "four":'4', "five":'5', "six":'6', "seven":'7', "eight":'8', "nine":'9' }
    temp = ''
    ret = ''
    for c in numbers:
        temp += c
        if temp in dic.keys():
            ret += dic[temp]
            temp = ''
    return int(ret)